airflow/task/task_runner/standard_task_runner.py

```diff
@@ -186,3 +193,12 @@ def get_process_pid(self) -> int:
         if self.process is None:
             raise RuntimeError("Process is not started yet")
         return self.process.pid
+
+    def _read_task_utilization(self):
+        while True:
+            dag_id = self._task_instance.dag_id
+            task_id = self._task_instance.task_id
```

Comment:
I guess we could grab it outside of loop

Reply:
100%. Good catch!